Source/DafnyCore/Compilers/Compiler-java.cs

```diff
@@ -1150,18 +1171,18 @@ protected class ClassWriter : IClassWriter {
       if (type != null && type.AsArrayType != null) {
         arrays.Add(type.AsArrayType.Dims);
       }
-      if (type.IsDatatype && type.AsDatatype is TupleTypeDecl tupleDecl) {
+      if (type != null && type.IsDatatype && type.AsDatatype is TupleTypeDecl tupleDecl) {
         tuples.Add(tupleDecl.NonGhostDims);
       }
-      if (type.IsTypeParameter) {
+      if (type != null && type.IsTypeParameter) {
         EmitSuppression(wr);
       }
       wr.Write("{0} {1}", type != null ? TypeName(type, wr, tok) : "Object", name);
       if (leaveRoomForRhs) {
         Contract.Assert(rhs == null); // follows from precondition
       } else if (rhs != null) {
         wr.WriteLine($" = {rhs};");
-      } else if (type.IsIntegerType) {
+      } else if (type != null && type.IsIntegerType) {
```

Comment:
When is type null here?

Reply:
It's used to represent the native java.lang.Object type. There's a big comment on `NativeObjectType`, which is what I should be using here instead of `null` :)